src/webhandlers.go

```diff
@@ -113,6 +113,57 @@ func getAverageColor(proportionsAndColors []proportionAndColor) ([3]int, error)
 	return ans, nil
 }
 
+// TODO: not forget to replace all the json fuck with `writeJSON` when fixing merge conflicts
+func HandleApiDays(w http.ResponseWriter, r *http.Request) {
+	session, _ := cookieStorage.Get(r, "session")
+	// Panics if user is not authorized. Will be fixed with the appropriate middleware
+	userID := session.Values["id"].(int)
+
+	panicIfError(r.ParseForm())
+
+	date, err := time.Parse("2006-01-02", r.FormValue("date"))
+	if err != nil {
+		js, err := json.Marshal(map[string]interface{}{"ok": false, "error": "Unable to parse date",
+			"error_type": "incorrect_date"})
+		panicIfError(err)
+		http.Error(w, string(js), http.StatusPreconditionFailed)
+		return
+	}
+
+	// TODO: the two lines below are formatted automatically this way. I wonder if it is possible to write it better
+	if len(r.Form["activity_type"]) != len(r.Form["activity_proportion"]) ||
+		len(r.Form["emotion_type"]) != len(r.Form["emotion_proportion"]) {
+
+		js, err := json.Marshal(map[string]interface{}{"ok": false,
```

Comment:
@DarkSquirrelComes, I am planning to request your complete review on the PR when I finish it, but I have a question so far. I hate how these lines look together (especially if I remove the blank line). Is there a nice way to fix it? Our hard limit is 120 characters per line (including the indentation, my tab size is 4)

I would also be thankful if you have checked other TODOs added in this PR and tell me if there is something you disagree with or know a better way for. Thank you!

Reply:
This autoformat seems good-looking to me. But I don't really understand, why you compare lens of this values?

Reply:
> This autoformat seems good-looking to me.

Then should I level the empty line there? Look what happens if I remove it:
```Golang
	// TODO: the two lines below are formatted automatically this way. I wonder if it is possible to write it better
	if len(r.Form["activity_type"]) != len(r.Form["activity_proportion"]) ||
		len(r.Form["emotion_type"]) != len(r.Form["emotion_proportion"]) {
		js, err := json.Marshal(map[string]interface{}{"ok": false,
```
I can't understand where the condition of this `if` ends and the body begins based on the indentation!


> But I don't really understand, why you compare lens of this values?

Each activity/emotion here is described with one type and one proportion. I ensure there exists a type for each proportion and a proportion for each value. You can see more details on this method in [the documentation](https://github.com/LifeCalendarTeam/life_calendar/blob/api_days/docs/api/README.md#apidays)